gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -160,17 +181,28 @@ public EntityResponse visit(DataFetcherNode dataFetcherNode) {
 
     // if the data fetcher node is fetching paginated records and the client has requested for
     // total, the total number of entities has to be fetched separately
+
+    EntityFetcherResponse response;
     if (dataFetcherNode.canFetchTotal()) {
       // since, the pagination is pushed down to the data store, total can be requested directly
       // from the data store. Request it async to parallelize with rest of entity fetch
       CompletableFuture<Long> totalFuture =
           CompletableFuture.supplyAsync(
-              () -> entityFetcher.getTotal(context, entitiesRequest), this.executorService);
-      return new EntityResponse(entityFetcher.getEntities(context, request), totalFuture.join());
+              () -> entityFetcher.getTotal(context, request), this.executorService);
+      EntityFetcherResponse entityFetcherResponse = entityFetcher.getEntities(context, request);
+      response =
+          childEntityFetcherResponse != null
+              ? intersectEntities(List.of(childEntityFetcherResponse, entityFetcherResponse))
+              : entityFetcherResponse;
+      return new EntityResponse(response, totalFuture.join());
     } else {
       // if the data fetcher node is not paginating, the total number of entities is equal to number
       // of records fetched
-      EntityFetcherResponse response = entityFetcher.getEntities(context, request);
+      EntityFetcherResponse entityFetcherResponse = entityFetcher.getEntities(context, request);
+      response =
+          childEntityFetcherResponse != null
+              ? intersectEntities(List.of(childEntityFetcherResponse, entityFetcherResponse))
```

Comment:
Curious Q: Do we still need to intersect when `childEntityFetcherResponse` is `not null`? In that case, wouldn't the child filter have already applied in subsequent query?

Reply:
`intersectEntities` also takes care of merging the selection attributes fetched from various sources

If we fetch attribute A from QS, and attribute B from EDS for the same entity, `intersectEntities` would merge the responses, and have both A and B added as an attribute on the entity response